### PR TITLE
Integration tests and multi-module support

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR brings multi-module and integration test support.

The test in `android` module allows us to verify that the test coverage works with a multi-module configuration.

The idea is to have a core version totally independent from Android SDK

Please @juanpedromoreno could you review? Thanks

### CODE REVIEW

File: android/src/main/scala/com/fortysevendeg/mvessel/package.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.fortysevendeg
+
+package object mvessel {
```
Comment (47degci) [inline android/src/main/scala/com/fortysevendeg/mvessel/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/47deg/mvessel/file/827260043/issues/source?bid=2442814&fileBranchId=2780198#l3)

File: core/src/it/scala/com/fortysevendeg/mvessel/DatabaseMetaDataSpec.scala [hunk 2]
```diff
@@ -17,9 +16,13 @@ trait DatabaseMetaDataSpecification
 
   sequential
 
-  var connection: Option[Connection] = None
+  var connection: Option[SQLConnection] = None
```
Comment (47degci) [inline core/src/it/scala/com/fortysevendeg/mvessel/DatabaseMetaDataSpec.scala:19]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/47deg/mvessel/file/827260045/issues/source?bid=2442814&fileBranchId=2780198#l19)

File: android/src/main/scala/com/fortysevendeg/mvessel/package.scala (hunk 1 above)
Comment (juanpedromoreno) [inline android/src/main/scala/com/fortysevendeg/mvessel/package.scala]: Maybe we could remove this package class.

File: project/Libraries.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import sbt._
+
+object Libraries {
+
+  lazy val specs2Core = "org.specs2" %% "specs2-core" % "3.3.1"
```
Comment (juanpedromoreno) [inline project/Libraries.scala]: There is a new version.

File: android/src/main/scala/com/fortysevendeg/mvessel/package.scala (hunk 1 above)
Comment (fedefernandez) [inline android/src/main/scala/com/fortysevendeg/mvessel/package.scala]: The `coverage` plugin doesn't work if there isn't source code in main

File: project/Settings.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    Classpaths.typesafeReleases,
+    DefaultMavenRepository,
+    Resolver.typesafeIvyRepo("snapshots"),
+    Resolver.sonatypeRepo("releases"),
+    Resolver.sonatypeRepo("snapshots"),
+    "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases"
+  )
+
+  lazy val basicSettings = Seq(
+    scalaVersion := "2.11.7",
```
Comment (juanpedromoreno) [inline project/Settings.scala]: Maybe you should move the scala version to the Versions file.

File: project/Settings.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    resolvers ++= commonResolvers,
+    logLevel := Level.Info,
+    scalacOptions in Compile ++= Seq("-deprecation", "-unchecked", "-feature"),
+    javacOptions in Compile ++= Seq("-target", "1.7", "-source", "1.7"),
+    javacOptions in Compile += "-deprecation",
```
Comment (juanpedromoreno) [inline project/Settings.scala]: Why two lines?

Comment (juanpedromoreno) [inline project/Settings.scala]: The core module should be compatible with java 1.8.

Comment (fedefernandez) [inline project/Settings.scala]: Need to be compatible with android :disappointed:

**Comment by juanpedromoreno** [comment]
LGTM!

**Pull request merged**